Framework/API/src/ComponentInfo.cpp

```diff
@@ -13,16 +13,39 @@ namespace API {
 /**
  * Constructor
  * @param componentInfo : Beamline wrapped ComponentInfo
- * @param componentIds : Component Ids ordered by component index
+ * @param componentIds : l-value reference to ComponentIDs ordered by component
+ * index
  */
-ComponentInfo::ComponentInfo(const Beamline::ComponentInfo &componentInfo,
-                             std::vector<Geometry::ComponentID> &&componentIds)
+ComponentInfo::ComponentInfo(
+    const Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> &componentIds)
     : m_componentInfo(componentInfo),
       m_componentIds(
           boost::make_shared<std::vector<Geometry::ComponentID>>(componentIds)),
       m_compIDToIndex(boost::make_shared<
           std::unordered_map<Geometry::IComponent *, size_t>>()) {
+  init();
+}
 
+/**
+ * Constructor
+ * @brief ComponentInfo::ComponentInfo
+ * @param componentInfo
+ * @param componentIds : r-value reference to ComponentIDs ordered by component
+ * index
+ */
+ComponentInfo::ComponentInfo(
+    const Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> &&componentIds)
+    : m_componentInfo(componentInfo),
```

Comment:
I'm not sure that that would work with the reference collapsing rules, or without adding a universal reference. I've added a `std::move`.  Let me know if you have a better idea.